lib/sshutils/server.go

```diff
@@ -565,6 +565,10 @@ func (s *Server) HandleConnection(conn net.Conn) {
 			for {
 				select {
 				case req := <-reqs:
+					if req == nil {
```

Comment:
Is a `nil` req something we expect to happen, or may we have bug somewhere else too?

(Definitely keep the change, I'm just curious on what prompted you to fix it.)

Reply:
I saw the panic during an integration test and went to investigate it.
The problem is caused [here](https://github.com/gravitational/crypto/blob/c3f983bc73dcf711908e49d91bda9cffbecb13cd/ssh/mux.go#L189-L191).
When an error is returned, the `loop` function closes the channels and the receiver receives a nil value and panics.

We also have another usage of the same channel outside of the error handling that does this check.